### ISSUE TITLE
BUG: not all required file blocks cause validation error when more than one file field and a hidden group are present in the form

### ISSUE THREAD

**Issue by pwkip** [pwkip/contact-form-7-conditional-fields#79]
When there are multiple `file*` fields in the form, and a hidden group, the file fields don't seem to trigger validation error.
```
[group g][text s][/group]
[file* f2]
[file* f1]
[submit]
```
![image](https://user-images.githubusercontent.com/1150196/117104378-4cbdf700-ad7c-11eb-9b72-30eb9354fdb4.png)
https://conditional-fields-cf7.bdwm.be/form-tester/?test-form=Github+issue+79

Without a hidden group present, it seems to work fine:
```
[file* f2]
[file* f1]
[submit]
```
![image](https://user-images.githubusercontent.com/1150196/117104935-4f6d1c00-ad7d-11eb-8e57-f10760ea4c45.png)

Possibly related to  this line:
```
add_filter( 'wpcf7_validate_file*', array($this, 'skip_validation_for_hidden_file_field'), 30, 3);
```

It was reported that commenting out this line, solve the problem.

Thanks @oselan for reporting this error and pointing out the possible cause!

**Comment by Oselan** [comment]
I didn't notice anything on my end.  I didn't commit it - I actually used remove filter in my theme functions.php but had to place the plugin class instance in a variable $cf7cf
 
global $cf7cf;
remove_filter( 'wpcf7_validate_file*', array($cf7cf,'skip_validation_for_hidden_file_field'), 30  );

**Comment by McSym** [comment]
I just commented out this line in plugin php file. Works well.

**Comment by pwkip** [comment]
(note to self) also add tests to make sure that attachments are sent correctly.

**Comment by pwkip** [comment]
fixed in version 2.0.4
